File: reverseCoding/lord-of-the-rings/gen.py
from random import randint
t=randint(1,10)
print(t)
while(t):
	t-=1
	n=randint(1,100)
	q=randint(1,100)
	print(n,q)
	for i in range(n):
		print(randint(1,1000),end=" ")
	print()
	for i in range(q):
		x=randint(1,n)
		y=randint(1,n)
		print(min(x,y),max(x,y),randint(1,1000))

File: reverseCoding/lord-of-the-rings/gen.cpp
#include<bits/stdc++.h>
using namespace std;
#define ll long long
int main(){
	ll t,n,q,x,y,i;
	t=1+rand()%static_cast<ll>(10);
	printf("%lld\n",t);
	while(t--){
		n=1+rand()%static_cast<ll>(100000);
		q=1+rand()%static_cast<ll>(100000);
		printf("%lld %lld\n",n,q);
		for(i=0;i<n;i++)
			printf("%lld ",1+rand()%static_cast<ll>(1000000));
		printf("\n");
		for(i=0;i<q;i++){
			x=1+rand()%static_cast<ll>(n);
			y=1+rand()%static_cast<ll>(n);
			printf("%lld %lld %lld\n",min(x,y),max(x,y),1+rand()%static_cast<ll>(1000000));
		}
	}
	return 0;
}